pkg/client/interfaces.go

```diff
@@ -133,3 +134,10 @@ func IgnoreNotFound(err error) error {
 	}
 	return err
 }
+
+// KubernetesResource allows functions to work indistinctly with any resource that
+// implements both Object interfaces.
+type KubernetesResource interface {
```

Comment:
Given that this is a union of two `*.Object` types would it be more intuitive for this to also include `Object` in the name?

Also, I wonder if the `client` package is the most appropriate place for this type. Nothing is really jumping out to me as an alternate though. Perhaps `pkg/controller/controllerutil`? :man_shrugging:

Reply:
Renamed and moved